src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java

```diff
@@ -50,30 +51,87 @@
  * </li>
  * </ul>
  * <p>
- * To prevent {@code FooCheck} violations from being reported write:
+ * To use default module configuration:
  * </p>
  * <pre>
- * &#64;SuppressWarnings("foo") interface I { }
- * &#64;SuppressWarnings("foo") enum E { }
- * &#64;SuppressWarnings("foo") InputSuppressWarningsFilter() { }
+ * &lt;module name=&quot;TreeWalker&quot;&gt;
+ *   &lt;module name=&quot;MemberName&quot;/&gt;
+ *   &lt;module name=&quot;ConstantName&quot;/&gt;
+ *   &lt;module name=&quot;ParameterNumber&quot;/&gt;
+ *   &lt;module name=&quot;NoWhitespaceAfter&quot;/&gt;
+ *
+ *   &lt;module name=&quot;SuppressWarningsHolder&quot;/&gt;
+ * &lt;/module&gt;
+ * &lt;module name=&quot;SuppressWarningsFilter&quot;/&gt;
+ * </pre>
+ * <pre>
+ * class Test {
+ *
+ *     private int K; // violation
+ *     &#64;SuppressWarnings({"membername"})
+ *     private int J; // ok
+ *
+ *     private static final int i = 0; // violation
+ *     &#64;SuppressWarnings("checkstyle:constantname")
+ *     private static final int m = 0; // ok
+ *
+ *     public void needsLotsOfParameters (int a, // violation
+ *       int b, int c, int d, int e, int f, int g, int h) {
+ *       // ...
+ *     }
+ *
+ *     &#64;SuppressWarnings("paramnum")
+ *     public void needsLotsOfParameters1 (int a, // ok
```

Comment:
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/13e36d4_2022234230/config_annotation.html#SuppressWarningsHolder_Examples
Not ok. I am getting a violation when documentation is saying I shouldn't be. Something is wrong in example.

See line 16 in output.

````
$ cat TestClass.java
class Test {

   private int K; // violation
   @SuppressWarnings({"membername"})
   private int J; // ok

   private static final int i = 0; // violation
   @SuppressWarnings("checkstyle:constantname")
   private static final int m = 0; // ok

   public void needsLotsOfParameters (int a, // violation
      int b, int c, int d, int e, int f, int g, int h) {
   }

   @SuppressWarnings("paramnum")
   public void needsLotsOfParameters1 (int a, // ok
      int b, int c, int d, int e, int f, int g, int h) {
   }

   private int [] ARR; // 2 violations
   @SuppressWarnings("all")
   private int [] ARRAY; // ok
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

<module name="TreeWalker">
  <module name="MemberName"/>
  <module name="ConstantName"/>
  <module name="ParameterNumber"/>
  <module name="NoWhitespaceAfter"/>

  <module name="SuppressWarningsHolder"/>
</module>
<module name="SuppressWarningsFilter"/>
</module>

$ java -jar checkstyle-10.3.1-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:3:16: Name 'K' must match pattern '^[a-z][a-zA-Z0-9]*$'. [MemberName]
[ERROR] TestClass.java:7:29: Name 'i' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [ConstantName]
[ERROR] TestClass.java:11:16: More than 7 parameters (found 8). [ParameterNumber]
[ERROR] TestClass.java:16:16: More than 7 parameters (found 8). [ParameterNumber]
[ERROR] TestClass.java:20:16: 'int' is followed by whitespace. [NoWhitespaceAfter]
[ERROR] TestClass.java:20:19: Name 'ARR' must match pattern '^[a-z][a-zA-Z0-9]*$'. [MemberName]
Audit done.
Checkstyle ends with 6 errors.
````

Other examples seem to be reporting as documented.

Reply:
Please look at this comment https://github.com/checkstyle/checkstyle/pull/11542#issuecomment-1126821468

Reply:
```
$ cat Test.java
class Test { 

  public void needsLotsOfParameters (int a, // violation
    int b, int c, int d, int e, int f, int g, int h) {
  }

 @SuppressWarnings("paramnum")
 public void needsLotsOfParameters1 (int a, // ok
    int b, int c, int d, int e, int f, int g, int h) {
 }
 
}

  $ cat Config.xml
class Test { 

  public void needsLotsOfParameters (int a, // violation
    int b, int c, int d, int e, int f, int g, int h) {
  }

 @SuppressWarnings("paramnum")
 public void needsLotsOfParameters1 (int a, // ok
    int b, int c, int d, int e, int f, int g, int h) {
 }
 
}

  kevin@kevin-Inspiron-15-5510:~/Desktop/C_Style$ java -jar /home/kevin/Downloads/checkstyle-10.3-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/kevin/Desktop/C_Style/Test.java:3:15: More than 7 parameters (found 8). [ParameterNumber]
Audit done.
Checkstyle ends with 1 errors.

```

Reply:
Your example you just provided is using a completely different config than the documentation provides which my example replicates.

The issue still stands the documentation doesn't line up and show a valid result.

Reply:
for now is there any changes that i can make